src/cli/subcommands/f3_cmd.rs

```diff
@@ -119,6 +129,67 @@ impl F3Commands {
             Self::Certs(cmd) => cmd.run(client).await,
             Self::PowerTable(cmd) => cmd.run(client).await,
+            Self::Ready { wait, threshold } => {
+                let is_running = client.call(F3IsRunning::request(())?).await?;
+                if !is_running {
+                    anyhow::bail!("F3 is not running");
+                }
+
+                async fn get_heads(
+                    client: &rpc::Client,
+                ) -> anyhow::Result<(Tipset, FinalityCertificate)> {
+                    let cert_head = client.call(F3GetLatestCertificate::request(())?).await?;
+                    let chain_head = client.call(ChainHead::request(())?).await?;
+                    Ok((chain_head, cert_head))
+                }
+
+                let ticker = Ticker::new(0.., Duration::from_secs(1));
+                let mut stdout = std::io::stdout();
+                let mut text = String::new();
+                for _ in ticker {
+                    match get_heads(&client).await {
+                        Ok((chain_head, cert_head)) => {
+                            if !text.is_empty() {
+                                write!(
+                                    stdout,
+                                    "\r{}{}",
+                                    anes::MoveCursorUp(1),
+                                    anes::ClearLine::All,
+                                )?;
+                            }
+                            if cert_head.chain_head().epoch + threshold as i64 >= chain_head.epoch()
+                            {
+                                text = format!(
+                                    "[+] F3 is in sync. Chain head epoch: {}, F3 head epoch: {}",
+                                    chain_head.epoch(),
+                                    cert_head.chain_head().epoch
+                                );
+                                println!("{text}");
+                                break;
+                            } else {
+                                text = format!(
+                                    "[-] F3 is not in sync. Chain head epoch: {}, F3 head epoch: {}",
+                                    chain_head.epoch(),
+                                    cert_head.chain_head().epoch
+                                );
+                                if !wait {
+                                    anyhow::bail!("{text}");
+                                } else {
+                                    println!("{text}");
+                                }
+                            }
+                        }
+                        Err(e) => {
+                            if !wait {
+                                anyhow::bail!("Failed to check F3 sync status: {e}");
+                            }
+                            // When waiting, log the error but continue
+                            eprintln!("Warning: Failed to fetch heads: {e}. Retrying...");
```

Comment:
@hanabi1224 We really should error out here after small number of retries otherwise we will get:

<img width="497" height="236" alt="Screenshot 2025-08-22 at 7 41 07 PM" src="https://github.com/user-attachments/assets/8bd8c13e-1ecd-4f34-915f-cee288838388" />

Reply:
Now it exits after 5 retries